File: mobile_app_bcp/lib/shared/models/solicitud_model.dart
// shared/models/solicitud_model.dart
class SolicitudModel {
  final String idSolicitud;
  final String? numeroExpediente;
  final String idCliente;
  final String? idNegocio;
  final String? idAsesor;
  final String? idProductoCredito;
  final String? canalOrigen;
  final double montoSolicitado;
  final double? montoAprobado;
  final int plazoMeses;
  final String moneda;
  final double? teaReferencial;
  final bool? conSeguroDesgravamen;
  final String? garantia;
  final String? destinoCredito;
  final double? cuotaEstimada;
  final String estado;
  final String? resultadoPreevaluacion;
  final int? puntajePreevaluacion;
  final String? resultadoBuro;
  final String? motivoRechazo;
  final DateTime? createdAt;

  SolicitudModel({
    required this.idSolicitud,
    this.numeroExpediente,
    required this.idCliente,
    this.idNegocio,
    this.idAsesor,
    this.idProductoCredito,
    this.canalOrigen,
    this.montoSolicitado = 0,
    this.montoAprobado,
    this.plazoMeses = 12,
    this.moneda = 'PEN',
    this.teaReferencial,
    this.conSeguroDesgravamen,
    this.garantia,
    this.destinoCredito,
    this.cuotaEstimada,
    this.estado = 'BORRADOR',
    this.resultadoPreevaluacion,
    this.puntajePreevaluacion,
    this.resultadoBuro,
    this.motivoRechazo,
    this.createdAt,
  });

  String get montoFormateado => 'S/ ${montoSolicitado.toStringAsFixed(2)}';

  factory SolicitudModel.fromJson(Map<String, dynamic> json) {
    return SolicitudModel(
      idSolicitud: json['id_solicitud'] ?? '',
      numeroExpediente: json['numero_expediente'],
      idCliente: json['id_cliente'] ?? '',
      idNegocio: json['id_negocio'],
      idAsesor: json['id_asesor'],
      idProductoCredito: json['id_producto_credito'],
      canalOrigen: json['canal_origen'],
      montoSolicitado: (json['monto_solicitado'] ?? 0).toDouble(),
      montoAprobado: json['monto_aprobado'] != null
          ? (json['monto_aprobado']).toDouble()
          : null,
      plazoMeses: json['plazo_meses'] ?? 12,
      moneda: json['moneda'] ?? 'PEN',
      teaReferencial: json['tea_referencial'] != null
          ? (json['tea_referencial']).toDouble()
          : null,
      conSeguroDesgravamen: json['con_seguro_desgravamen'],
      garantia: json['garantia'],
      destinoCredito: json['destino_credito'],
      cuotaEstimada: json['cuota_estimada'] != null
          ? (json['cuota_estimada']).toDouble()
          : null,
      estado: json['estado'] ?? 'BORRADOR',
      resultadoPreevaluacion: json['resultado_preevaluacion'],
      puntajePreevaluacion: json['puntaje_preevaluacion'],
      resultadoBuro: json['resultado_buro'],
      motivoRechazo: json['motivo_rechazo'],
      createdAt: json['created_at'] != null
          ? DateTime.tryParse(json['created_at'])
          : null,
    );
  }
}

File: backend_core_mobile/app/models/solicitud_model.py
# solicitud_model.py
import uuid
import datetime
from sqlalchemy import Column, String, Integer, DateTime, ForeignKey, Numeric, Boolean
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from app.database.session import Base

class ProductoCredito(Base):
    __tablename__ = "productos_credito"

    id_producto_credito = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    codigo = Column(String(30), unique=True, nullable=False)
    nombre = Column(String(120), nullable=False)
    tipo = Column(String(50))
    tea_con_seguro = Column(Numeric(5, 2), nullable=False)
    tea_sin_seguro = Column(Numeric(5, 2), nullable=False)
    monto_minimo = Column(Numeric(12, 2), nullable=False)
    monto_maximo = Column(Numeric(12, 2), nullable=False)
    plazo_minimo = Column(Integer, nullable=False)
    plazo_maximo = Column(Integer, nullable=False)
    moneda = Column(String(3), default="PEN")
    estado = Column(String(20), default="ACTIVO")
    created_at = Column(DateTime(timezone=True))

class SolicitudCredito(Base):
    __tablename__ = "solicitudes_credito"

    id_solicitud = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    numero_expediente = Column(String(30), unique=True, nullable=False)
    id_cliente = Column(UUID(as_uuid=True), ForeignKey("clientes.id_cliente"), nullable=False)
    id_negocio = Column(UUID(as_uuid=True), ForeignKey("negocios_cliente.id_negocio"), nullable=False)
    id_asesor = Column(UUID(as_uuid=True), ForeignKey("asesores.id_asesor"), nullable=True)
    id_producto_credito = Column(UUID(as_uuid=True), ForeignKey("productos_credito.id_producto_credito"), nullable=False)
    canal_origen = Column(String(30), nullable=False)
    monto_solicitado = Column(Numeric(12, 2), nullable=False)
    monto_aprobado = Column(Numeric(12, 2), nullable=True)
    plazo_meses = Column(Integer, nullable=False)
    moneda = Column(String(3), default="PEN")
    tea_referencial = Column(Numeric(5, 2), nullable=False)
    con_seguro_desgravamen = Column(Boolean, default=True)
    garantia = Column(String(50))
    destino_credito = Column(String)
    cuota_estimada = Column(Numeric(12, 2))
    estado = Column(String(30), nullable=False)
    resultado_preevaluacion = Column(String(30))
    puntaje_preevaluacion = Column(Integer)
    resultado_buro = Column(String(30))
    motivo_rechazo = Column(String)
    condicion_adicional = Column(String)
    firma_cliente_base64 = Column(String)
    lat_captura = Column(Numeric(10, 7))
    lng_captura = Column(Numeric(10, 7))
    pendiente_sync = Column(Boolean, default=False)
    created_at = Column(DateTime(timezone=True))
    updated_at = Column(DateTime(timezone=True))

    cliente = relationship("Cliente", foreign_keys=[id_cliente])
    negocio = relationship("NegocioCliente", foreign_keys=[id_negocio])
    asesor = relationship("Asesor", foreign_keys=[id_asesor])
    producto = relationship("ProductoCredito", foreign_keys=[id_producto_credito])

class SolicitudNotaInterna(Base):
    __tablename__ = "solicitudes_notas_internas"

    id_nota = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    id_solicitud = Column(UUID(as_uuid=True), ForeignKey("solicitudes_credito.id_solicitud"), nullable=False)
    id_asesor = Column(UUID(as_uuid=True), ForeignKey("asesores.id_asesor"), nullable=True)
    contenido = Column(String(500), nullable=False)
    created_at = Column(DateTime(timezone=True), default=datetime.datetime.now)

    solicitud = relationship("SolicitudCredito", foreign_keys=[id_solicitud])
    asesor = relationship("Asesor", foreign_keys=[id_asesor])

